frame/contracts/src/chain_extension.rs

```diff
@@ -112,20 +112,48 @@ pub trait ChainExtension<C: Config> {
 	}
 }
 
-/// Implementation that indicates that no chain extension is available.
-impl<C: Config> ChainExtension<C> for () {
-	fn call<E>(_func_id: u32, mut _env: Environment<E, InitState>) -> Result<RetVal>
+/// A [`ChainExtension`] that can be composed with other extensions using a tuple.
+///
+/// An extension that implements this trait can be put in a tuple in order to have multiple
+/// extensions available. The tuple implementation routes requests based on the first two
+/// MSB bytes of the `func_id` pased to `call`.
+///
+/// # Note
+///
+/// Register your id with the [chain extension registry](https://github.com/paritytech/ChainExtension-registry)
+/// to ensure that there are
+/// no collisions with other vendors.
+pub trait RegisteredChainExtension<C: Config>: ChainExtension<C> {
+	/// The extensions globally unique identifier.
+	const ID: u16;
+}
+
+#[impl_trait_for_tuples::impl_for_tuples(10)]
```

Comment:
So the maximum number of chain extensions than can be composed is ten? We should document this limitation.

Reply:
Yes. However, this number is completely arbitrary. I have no idea what a good number is.

Reply:
Added some more docs including this.